nixos/tests/rsyslogd.nix

```diff
@@ -0,0 +1,27 @@
+import ./make-test.nix ({ pkgs, lib, ... }:
+{
+  name = "rsyslogd";
+  meta.maintainers = [ lib.maintainers.aanderse ];
+
+  nodes = {
+    machine1 =
+      { config, pkgs, ... }:
+      { services.rsyslogd.enable = true;
+        services.journald.forwardToSyslog = false;
+      };
+
+    machine2 =
+      { config, pkgs, ... }:
+      { services.rsyslogd.enable = true;
+      };
```

Comment:
I would rather make two test, so the tests can be started independently.

Reply:
To do so a test file can return a set of tests.

Reply:
After going through the examples I converted my file into a set of multiple tests, though I can't seem to run the tests interactively. What would you suggest as the most basic test file to look at which has multiple tests in it?

Reply:
Take elasticsearch as an example:

```console
$ nix-build nixos/tests/elk.nix -A ELK-5
```

Reply:
Here is the test script I came up with:

```
{ system ? builtins.currentSystem }:

with import ../lib/testing.nix { inherit system; };
{
  test1 = makeTest {
    name = "rsyslogd-test1";
    meta.maintainers = [ lib.maintainers.aanderse ];

    machine =
      { config, pkgs, ... }:
      { services.rsyslogd.enable = true;
        services.journald.forwardToSyslog = false;
      };

    # ensure rsyslogd isn't receiving messages from journald if explicitly disabled
    testScript = ''
      $machine->waitForUnit("default.target");
      $machine->fail("test -f /var/log/messages");
    '';
  };

  test2 = makeTest {
    name = "rsyslogd-test2";
    meta.maintainers = [ lib.maintainers.aanderse ];

    machine =
      { config, pkgs, ... }:
      { services.rsyslogd.enable = true;
      };

    # ensure rsyslogd is receiving messages from journald
    testScript = ''
      $machine->waitForUnit("default.target");
      $machine->succeed("test -f /var/log/messages");
    '';
  };
}
```

When I run this `nix-build rsyslogd.nix -A test1` the result/ folder has no bin/ under it, so I cannot interactively run the test in a VM. Reviewing elk.nix was helpful, but I'm still not sure how I can interactively run the test as described in https://nixos.org/nixos/manual/index.html#sec-running-nixos-tests.

Reply:
I miss-understood you. Just use:

```console
$ nix-build nixos/tests/elk.nix -A ELK-5.driver   
```